instrumentation/mongodb/src/main/java/brave/mongodb/TraceMongoCommandListener.java

```diff
@@ -0,0 +1,136 @@
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.mongodb;
+
+import brave.Span;
+import brave.internal.Nullable;
+import brave.propagation.ThreadLocalSpan;
+import com.mongodb.MongoException;
+import com.mongodb.MongoSocketException;
+import com.mongodb.connection.ConnectionDescription;
+import com.mongodb.connection.ConnectionId;
+import com.mongodb.event.CommandFailedEvent;
+import com.mongodb.event.CommandListener;
+import com.mongodb.event.CommandStartedEvent;
+import com.mongodb.event.CommandSucceededEvent;
+import org.bson.BsonDocument;
+import org.bson.BsonValue;
+
+import java.net.InetSocketAddress;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * A MongoDB command listener that will report via Brave how long each command takes and other information about the
+ * commands.
+ *
+ * See <a href="https://github.com/openzipkin/brave/blob/master/instrumentation/mongodb/RATIONALE.md">RATIONALE.md</a>
+ * for implementation notes.
+ */
+final class TraceMongoCommandListener implements CommandListener {
+  // See https://docs.mongodb.com/manual/reference/command for the command reference
+  static final Set<String> COMMANDS_WITH_COLLECTION_NAME = new HashSet<>(Arrays.asList("aggregate", "count", "distinct",
+    "mapReduce", "geoSearch", "delete", "find", "findAndModify", "insert", "update", "collMod", "compact",
+    "convertToCapped", "create", "createIndexes", "drop", "dropIndexes", "killCursors", "listIndexes", "reIndex"));
+
+  final ThreadLocalSpan threadLocalSpan;
+
+  TraceMongoCommandListener(MongoDBTracing mongoDBTracing) {
+    this(ThreadLocalSpan.create(mongoDBTracing.tracing.tracer()));
+  }
+
+  TraceMongoCommandListener(ThreadLocalSpan threadLocalSpan) {
+    this.threadLocalSpan = threadLocalSpan;
+  }
+
+  /**
+   * Uses {@link ThreadLocalSpan} as there's no attribute namespace shared between callbacks, but
+   * all callbacks happen on the same thread.
+   */
+  @Override public void commandStarted(CommandStartedEvent event) {
+    Span span = threadLocalSpan.next();
+    if (span == null || span.isNoop()) return;
+
+    String commandName = event.getCommandName();
+    String databaseName = event.getDatabaseName();
+    BsonDocument command = event.getCommand();
+    String collectionName = getCollectionName(command, commandName);
+
+    span.name(getSpanName(commandName, collectionName))
+      .kind(Span.Kind.CLIENT)
+      .remoteServiceName("mongodb-" + databaseName)
+      .tag("mongodb.command", commandName);
```

Comment:
reconsider if this should be "mongodb.command_name" as it gets out of the way of "mongodb.command" if folks later add the full command as a property..

Reply:
I changed it to "command" to align with [MongoMetricsCommandListener](https://github.com/micrometer-metrics/micrometer/blob/master/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/mongodb/MongoMetricsCommandListener.java#L64). I figured there is value in being consistent there, and if we bring back the full command in the future, we can call that "full_command" or something like that.